### PR TITLE
updated docs for new CLI resources

### PR DESCRIPTION
**Pull request by epinzur**
* also moved `sync` command stuff to its own file.

### CODE REVIEW

File: docs-src/modules/developing/pages/specFiles.adoc [hunk 1]
```diff
@@ -0,0 +1,241 @@
+:tabs-sync-option:
+
+= Spec Files 
+
+In addition to modifiying resources directly, you can also use a *spec file* to describe the desired state of Kaskada.
+
+A spec file is a YAML file describing a set of Kaskada resources, for examples tables and views.
+
+The `sync` command is used to update the state of your system to match the file.
```
Comment (therapon) [inline docs-src/modules/developing/pages/specFiles.adoc:9]:
```suggestion
The `sync` command is used to update the state of your Kaskada system to match the resource descriptions in the spec file. 
```

**Review by kerinin — APPROVED** [review]
Might also want to update the CLI quickstart to show the imperative table creation command.

File: docs-src/modules/developing/pages/specFiles.adoc [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+
+The `sync` command is used to update the state of your system to match the file. 
+
+[NOTE]
+====
+Currently `sync` is only supported in the CLI client.
+====
+
+== Spec File Format
+
```
Comment (therapon) [inline docs-src/modules/developing/pages/specFiles.adoc:17]: Maybe add the list of resources that the spec file understands, as a list or a small table of contents. Then each one gets a subsection.

**Comment by kerinin** [comment]
May want to wait for this PR to merge before changing the quickstart tho: https://github.com/kaskada-ai/kaskada/pull/311

File: docs-src/modules/developing/pages/specFiles.adoc [hunk 3]
```diff
@@ -0,0 +1,241 @@
+:tabs-sync-option:
+
+= Spec Files 
+
+In addition to modifiying resources directly, you can also use a *spec file* to describe the desired state of Kaskada.
```
Comment (kevinjnguyen) [inline docs-src/modules/developing/pages/specFiles.adoc:5]: Do we intend to include the actual spec of each resource? For example, the [argument references](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/s3_bucket#argument-reference) for AWS S3 Buckets describe each of the fields/requirements.

File: docs-src/modules/developing/pages/specFiles.adoc [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+A spec file is a YAML file describing a set of Kaskada resources, for examples tables and views.
+
+The `sync` command is used to update the state of your system to match the file. 
+
+[NOTE]
+====
+Currently `sync` is only supported in the CLI client.
+====
+
+== Spec File Format
```
Comment (therapon) [inline docs-src/modules/developing/pages/specFiles.adoc:16]: Is it tables, views and materializations only that are supported? 
What I think I am missing is the spec for the spec file :) 
What are all the top level keys that the spec supports?

File: docs-src/modules/developing/pages/specFiles.adoc [hunk 5]
```diff
@@ -0,0 +1,241 @@
...
+  destination:
+    objectStore:
+      fileType: FILE_TYPE_PARQUET
+      outputPrefixUri: s3://my-bucket/materialization-output-prefix/
+  slice: {}
+----
+
+== Updating Kaskada to reflect the contents of a spec file
+
+To upteda a resource (table, view, or materialization), you first modify the resource in your spec file,
```
Comment (therapon) [inline docs-src/modules/developing/pages/specFiles.adoc:196]:
```suggestion
To update a resource (table, view, or materialization), you first modify the resource in your spec file, 
```

File: docs-src/modules/developing/pages/specFiles.adoc [hunk 6]
```diff
@@ -0,0 +1,241 @@
...
+  slice: {}
+----
+
+== Updating Kaskada to reflect the contents of a spec file
+
+To upteda a resource (table, view, or materialization), you first modify the resource in your spec file, 
+then use the `spec plan` command to preview the changes that will be made to the system. To make the 
+acutal changes, use the `spec apply` command.
+
+When a spec file is update, the CLI inspects all of the server's resources and all of the resources
```
Comment (therapon) [inline docs-src/modules/developing/pages/specFiles.adoc:200]:
```suggestion
When a spec file is updated, the CLI inspects all of the server's resources and all of the resources
```

**Review by kevinjnguyen — CHANGES_REQUESTED** [review]

File: docs-src/modules/developing/pages/specFiles.adoc [hunk 7]
```diff
@@ -0,0 +1,241 @@
...
+defined in your spec file, then takes whatever actions are necessary to reconcile the server's state.
+Applying a spec can create new resources, or update resources by deleting them & then recreating them.
+
+Note, if you remove a resource from a spec file, it will not be deleted from the system. Instead you 
+must delete those resources using the standard delete commands.
```
Comment (therapon) [inline docs-src/modules/developing/pages/specFiles.adoc:205]: Maybe link to the docs that have the "delete commands"

File: docs-src/modules/developing/pages/specFiles.adoc [hunk 8]
```diff
@@ -0,0 +1,241 @@
...
+
+[WARNING]
+.Table updates are destructive
+====
+Tables are currently immutable. 
+When the CLI updates a table, it does so by deleting the table and re-creating it.
+When this happens, all data previously loaded into the table is lost.
+====
+
+=== Previewing the changes
```
Comment (therapon) [inline docs-src/modules/developing/pages/specFiles.adoc:215]: Maybe move the text on line 227 here. A section followed immediately by an example comes across as a little bare.

File: docs-src/modules/developing/pages/tables.adoc [hunk 9]
```diff
@@ -128,22 +146,20 @@ You can delete a table using its name:
...
 [source,python]
-.with Python:
 ----
 from kaskada import table
 from kaskada.api.session import LocalBuilder
 
 session = LocalBuilder().build()
 
-table.delete_table(table = "Purchase")
+table.delete_table("Purchase")
```
Comment (therapon) [inline docs-src/modules/developing/pages/tables.adoc:155]: Any reason for removing the argument name?

File: docs-src/modules/developing/pages/views.adoc [hunk 10]
```diff
@@ -141,106 +181,39 @@ session = LocalBuilder().build()
...
-        } 
-      # A name used to refer to the view in queries
-    - view_name: CountPurchase
-      # The expression to substitute anywhere the view's name is used
-      expression: count(Purchase)
+kaskada-cli view delete PurchaseStats
 ----
+====
 
+A failed precondition error is returned if another view and/or
```
Comment (therapon) [inline docs-src/modules/developing/pages/views.adoc:192]:
```suggestion
A failed precondition error is returned when another view and/or
```

File: docs-src/modules/developing/pages/views.adoc [hunk 11]
```diff
@@ -141,106 +181,39 @@ session = LocalBuilder().build()
...
+materialization references the view. To continue with the deletion of
+the view, delete the dependent resources or supply the `force` flag to
```
Comment (therapon) [inline docs-src/modules/developing/pages/views.adoc:194]:
```suggestion
materialization references the view. To continue with the deletion of
the view we can either, delete the dependent resources manually with another delete operation, or, supply the `force` flag to
```

File: docs-src/modules/developing/pages/views.adoc [hunk 12]
```diff
@@ -141,106 +181,39 @@ session = LocalBuilder().build()
...
+delete the view forcefully. Forcefully deleting a view without
+deleting the dependent resources may result in the dependent resources
+functioning incorrectly.
```
Comment (therapon) [inline docs-src/modules/developing/pages/views.adoc:197]:
```suggestion
delete the view forcefully. Forcefully deleting a view without
deleting its dependent resources may result in its dependent resources
functioning incorrectly.
```

Comment (therapon) [reply docs-src/modules/developing/pages/views.adoc:197]: I might move the last sentence into an asciidoctor warning block
https://docs.asciidoctor.org/asciidoc/latest/syntax-quick-reference/#admonitions

File: docs-src/modules/developing/pages/specFiles.adoc (hunk 4 above)
Comment (epinzur) [reply docs-src/modules/developing/pages/specFiles.adoc:16]: yes, only tables, views, & materializations are supported.

File: docs-src/modules/developing/pages/tables.adoc (hunk 9 above)
Comment (epinzur) [reply docs-src/modules/developing/pages/tables.adoc:155]: Just trying to make things consistent.  the view & materialization docs weren't using argument names.

File: docs-src/modules/developing/pages/specFiles.adoc (hunk 3 above)
Comment (epinzur) [reply docs-src/modules/developing/pages/specFiles.adoc:5]: Eventually maybe yes, but here I just cut & pasted the text and format over from the old specFile sections in the `table` and `view` docs and made the minimum changes. I'd prefer to make larger SpecFile doc changes in a separate PR.

File: docs-src/modules/developing/pages/materializations.adoc [hunk 13]
```diff
@@ -107,30 +89,59 @@ materialization.create_materialization(
...
+----
+kaskada-cli materialization create PurchaseStats \
+    "{time: Purchase.purchase_time,entity: Purchase.customer_id,max_amount: Purchase.amount | max(),min_amount: Purchase.amount | min()}" \
+    --path-uri "file:///path/on/your/machine/"   
+----
+====
+
+[NOTE]
+====
+Currently the CLI can only create materializaitons that use the *Object Store* destination type.
```
[the remark this answers is not in the document]
Comment (epinzur) [reply docs-src/modules/developing/pages/materializations.adoc]: added note that `path-uri` does this

**Pull request merged**